### DISCUSSION TITLE
Google SAML/SSO

### DISCUSSION THREAD

**Discussion by P0531don** [snipe/snipe-it#12424] [Q&A] [answered]
New to Snipe-it.

Not finding any great documentation on Google integration.  I have the documentation on setting up SAML, however, my question is does this automatically import all of our users in Google Workspace? Or, just allow them to sign in using their Google account.  Do I need to pre-populate the users from a csv download from Google Workspace?

Thanks in advanced.

**Comment by Kaphalor** [comment]
According to our Infrastructure Teamlead, it would be best to create an LDAP connection to your Google Workspace (we did that few days ago). 
After the connection was established, you can synchronise LDAP to populate the users table in the database.
After that, you can add the SAML connection, which is what we will do either today or next week as well.

**Reply by P0531don** [reply]
I already have Google Workspace populated and don't plan to connect LDAP to it.  I was curious if the SAML connection automatically made accounts in snipe-it.

**Comment by Kaphalor** [comment, marked as answer]
Just-in-Time (JIT) Provisioning

SAML can only authenticate a user in Snipe-IT if the user exists in
Snipe-IT. We do not attempt to create a user on SAML login. This means that
if a user exists in your SAML directory but NOT in Snipe-IT, that user will
not be able to login to Snipe-IT with SAML. We typically recommend using
the LDAP <https://snipe-it.readme.io/docs/ldap-sync-login> sync to pull
your users in, then SAML to authenticate them.
https://snipe-it.readme.io/docs/saml

**Reply by P0531don** [reply]
Thank you for the info, I appreciate it.
